raft.go

```diff
@@ -1907,6 +2044,84 @@ func (r *Raft) electSelf() <-chan *voteResult {
 	return respCh
 }
 
+// preElectSelf is used to send a RequestPreVote RPC to all peers, and vote for
+// ourself. This will not increment the current term. The
+// response channel returned is used to wait for all the responses (including a
+// vote for ourself).
+// This must only be called from the main thread.
+func (r *Raft) preElectSelf() <-chan *preVoteResult {
+	// Create a response channel
+	respCh := make(chan *preVoteResult, len(r.configurations.latest.Servers))
+
+	// Increment the term
+	newTerm := r.getCurrentTerm() + 1
+
+	// Construct the request
+	lastIdx, lastTerm := r.getLastEntry()
+	req := &RequestPreVoteRequest{
+		RPCHeader: r.getRPCHeader(),
+		Term:      newTerm,
+		// this is needed for retro compatibility, before RPCHeader.Addr was added
+		Candidate:    r.trans.EncodePeer(r.localID, r.localAddr),
```

Comment:
```suggestion
```

This change goes in tandem with my other proposal to remove the Candidate field from the struct definition.